pkg/profiler/profiler.go

```diff
@@ -269,10 +274,15 @@ func (p *CgroupProfiler) Run(ctx context.Context) error {
 
 		// Because this is fd based, even if our program crashes or is ended
 		// without proper shutdown, things get cleaned up appropriately.
-		// TODO(brancz): destroy the returned link via bpf_link__destroy
-		if _, err := prog.AttachPerfEvent(fd); err != nil {
+		link, err := prog.AttachPerfEvent(fd)
+		if err != nil {
 			return fmt.Errorf("attach perf event: %w", err)
 		}
+		defer func() {
```

Comment:
This might be potentially the same issue as above, only that slightly less dangerous I think 😄